mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb

```diff
@@ -200,6 +218,168 @@ func resourceDataprocCluster() *schema.Resource {
+							AtLeastOneOf: clusterConfigKeys,
+							Computed:     true,
+							MaxItems:     1,
+							Description:  `The Dataproc cluster config for a cluster that does not directly control the underlying compute resources, such as a Dataproc-on-GKE cluster.`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+
+									"image_version": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										Default:          "3.1-dataproc-7",
+										AtLeastOneOf:     gkeClusterConfigKeys,
+										ForceNew:         true,
+										DiffSuppressFunc: dataprocImageVersionDiffSuppress,
+										Description:      `The Cloud Dataproc image version to use for the cluster - this controls the sets of software versions installed onto the nodes when you create clusters. If not specified, defaults to the latest version.`,
+									},
+
+									"optional_components": {
+										Type:        schema.TypeMap,
+										Optional:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Computed:    true,
+										Description: `The components that should be installed in this Dataproc cluster. The key must be a string from the KubernetesComponent enumeration. The value is the version of the software to be installed.`,
+									},
+
+									"properties": {
+										Type:        schema.TypeMap,
+										Optional:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Computed:    true,
+										Description: `The properties to set on daemon config files.`,
+									},
+
+									"namespace": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Default:      false,
+										AtLeastOneOf: gkeClusterConfigKeys,
+										ForceNew:     true,
+										Description:  `A namespace within the Kubernetes cluster to deploy into. If this namespace does not exist, it is created. If it exists, Dataproc verifies that another Dataproc VirtualCluster is not installed into it. If not specified, the name of the Dataproc Cluster is used.`,
+									},
+
+									"cluster": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Default:      false,
+										AtLeastOneOf: gkeClusterConfigKeys,
+										ForceNew:     true,
+										Description:  `A target GKE cluster to deploy to. It must be in the same project and region as the Dataproc cluster (the GKE cluster can be zonal or regional). Format: 'projects/{project}/locations/{location}/clusters/{cluster_id}'`,
+									},
+
+									"auxiliary_services": {
+										Type:         schema.TypeList,
+										Optional:     true,
+										Computed:     true,
+										MaxItems:     1,
+										AtLeastOneOf: gkeClusterConfigKeys,
+										Description:  `Auxiliary services configuration for a Cluster.`,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+
+												"metastore_service": {
```

Comment:
Child fields that are not `ForceNew` imply that they are updatable. If this is the case, a test step showing the updatability of these fields is preferred

Reply:
Thanks for the hint! Will add tests to mentioned child fields